### PR TITLE
feat: report templates + per-meeting report generation, versions & sidecar storage

### PR DESCRIPTION
**Pull request by Optic00**
> **Stacked on #250.** This branch is based on `feat/map-reduce-summarization-188`, so until #250 merges the diff below also includes #250's commits. The work specific to this PR is the report-storage-sidecar + templates + reports commits on top. Please review/merge after #250.

## What this does

Adds user-defined **report templates** and **per-meeting report generation/versioning**, persisted via a meeting **sidecar** — so a meeting can have more than one summary (the structured Standard plus any number of template-driven reports), switchable in the UI.

### Templates
- Built-in registry + pure template helpers; a locked **Standard** template plus a one-time seeded sample.
- Config storage with full **CRUD**, a configurable **default template**, and hardened `validate_template` (type + size limits, language-code validation).
- **CLI**: `list-templates` / `save-template` / `delete-template` / `set-default-template` / `reset-template`.
- **IPC bridge** (main handlers + preload + typed group) and a **Settings → Templates** tab with a `useTemplates` hook.
- Full-width, taller **prompt editor** (the shared `Textarea` no longer inherits the single-line input height, so `rows`/`min-h` apply).

### Reports
- Free-form template-generation path in the summarizer; `generate-report` CLI + IPC + `reports[]` model.
- **Report switcher** in MeetingDetail: generate from any template, switch between versions (shows model + date), set-active, delete; markdown render.
- **Versions + #249 snapshot**: reprocess snapshots the prior Standard report instead of overwriting it; note backed up with any content.
- **Sidecar storage** (`.md` + `.json`) with a format-agnostic meeting reader; `get-meeting` merges the sidecar.
- **Auto-generate** the default template's report for new recordings (heartbeated; never raises on import).
- Edge cases surfaced cleanly: unknown template / empty report → `STREAM_ERROR` (no phantom success).

## Tests

- **T2 e2e** (model-free, in CI): template CRUD + default + locked-Standard round-trips; `generate-report` persists into `reports[]` and sets active; generate/switch/delete on a real `.md` meeting via the sidecar; reprocess backup + set-active + delete round-trip.
- Renderer `typecheck` + `lint` green.
- Manual (macOS): Templates tab create/edit/delete + default; per-recording auto-report; report switcher generate/switch/delete; reprocess snapshot; sidecar files on disk.

---
## Summary by cubic
Adds report templates and per‑meeting report generation with versioning, stored in a `_reports.json` sidecar and switchable in the UI. Also ships a Templates settings tab + CLI, provider‑agnostic template streaming, scoped progress, auto‑generation of a default report for new recordings, and stronger validation/error handling.

- **New Features**
  - Templates: built‑in locked Standard + one‑time seeded sample; full CRUD in Settings and CLI (delete confirmation, reset, set default); hardened validation and on‑load normalization/repair; editor remounts per template and the prompt textarea now grows; CLI surfaces structured errors for locked templates.
  - Reporting: generate from any template with provider‑agnostic streaming (local or cloud); versions with switch/set‑active/delete (confirmations) and Markdown render; previous Standard snapshot is saved on reprocess; failures are tagged as report events and surfaced separately; `STREAM_COMPLETE` fires after saving to prevent stale reads.
  - Storage/IO: `_reports.json` sidecar for both `.md` and `.json` meetings; `get-meeting` merges the sidecar; reads are path‑contained; sidecar loads normalize key types and handle malformed data; set‑active/delete require the meeting to exist.
  - UX/Processing: progress events include `summaryFile` to scope map‑reduce status; the Processing screen ignores scoped progress from other meetings; unknown/empty results emit `STREAM_ERROR` with safe retry; summarization disables reasoning (`think=false`) with a one‑time fallback; meetings list omits transcripts for speed; Meeting detail lazy‑loads via a secure `get-meeting`; default‑template report auto‑generates for new recordings without blocking import.

- **Dependencies**
  - Require Python `ollama` >= `0.5.0` (enables `think=false`; older servers auto‑retry without it) and add support for `gemma4:4b`.

<sup>Written for commit 1e59188aa18d9715d8956a62005075782bc4cc94. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/ruzin/stenoai/pull/252?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by Optic00** [comment]
## cubic review — resolutions

Worked through all 12 findings. Fixes are in `0408444` (backend hardening + tests) and `32ee033` (renderer / report-vs-reprocess labelling). Each was independently re-reviewed.

**Fixed**
- **P1 `templates.py` format guard** — `validate_template` now type-guards `format` before the set-membership check, so a non-string/unhashable payload returns an error tuple instead of raising. Regression test added (4 malformed cases).
- **P1 `config.py` template seeding** — `_seed_sample_template` normalizes `custom_templates` (drops non-dict entries, tolerates a non-list value) so a malformed-but-parseable config can't crash startup. Resilience tests added.
- **P1 `ipc-contract.md` processing-progress** — payload documented as `{ line, summaryFile? }`.
- **P1 `summarizer.py` template path** — already addressed in `2033107` (cloud/adapter providers now stream the free-form template report; covered by the cloud-path test).
- **P2 `Settings.tsx` editor state** — `TemplateEditor` now remounts per template id (`key`), so switching the edit target can't carry stale form state into another template.
- **P2 `simple_recorder.py` set-active-report** — both `set-active-report` and `delete-report` now require the summary file to exist before touching the sidecar (no orphan sidecar state).
- **P2 `MeetingDetail.tsx` generate-report failure** — added an `onError` rollback for failures that occur before the first stream event, so the UI no longer strands in the analyzing state.
- **P3 `MeetingDetail.tsx` failure labelling** — report-generation failures are no longer shown as reprocess/model-memory failures. Both flows emit structurally identical `summary-complete`/`processing-complete` events, so discriminating via local state was order-sensitive; `main.js` now tags all generate-report terminal events with `report: true` and the renderer branches on the event flag (order-independent) on both the `summary-complete` and `processing-complete` failure paths.
- **P3 `test_report_store.py`** — tightened the absent-notes assertion to `assertIsNone` (was accepting `""`).
- **P0 `test_summarizer_template.py`** — construction now runs under a mocked `_ensure_ollama_ready`, so the tests no longer touch Ollama at construction time.

**Won't fix (with rationale)**
- **P2 `Processing.tsx` progress scoping** — the Processing screen is the active-recording flow; it does not yet have the meeting's `summaryFile` at the point progress arrives, and the concrete leak case (an old meeting's concurrent reprocess into an open note) is already scoped in `MeetingDetail.tsx`. Scoping the Processing badge would need a different signal and has little practical payoff. Leaving as-is.
- **P2 `report_store.py` sidecar normalization** — `load_sidecar` only sets defaults; it would only matter for a hand-corrupted sidecar (not a path the app produces), so the added type-normalization isn't worth the surface. Happy to add it if you'd prefer defensive parity with the config path.

Verification: full Python suite green (272), renderer typecheck clean, lint 0 errors.

**Comment by Optic00** [comment]
## cubic follow-up P2 — fixed in `5ea87ed`

Valid finding. My earlier fix normalized `custom_templates` only inside `_seed_sample_template`, which early-returns once `templates_seeded` is set — so an **already-seeded** config with malformed template state was never repaired and would crash later template reads/writes.

**Fix:** pulled normalization out of the seed gate into an ungated `_normalize_templates()` that runs on **every** load (before seeding). It coerces `custom_templates` to a list-of-dicts (dropping junk entries / a non-list value) and `template_overrides` to a dict-of-dicts. Repair is in-memory and persists on the next `_save()`; fresh-seed behavior is unchanged (`_normalize_templates` → `_seed_sample_template` still appends the sample once).

A Codex pass on the diff caught one extra gap beyond cubic's note: a `template_overrides` that *is* a dict but has a malformed per-template value (e.g. `{"standard": "oops"}`) still crashed `merge_templates` — now filtered too.

Tests: added `AlreadySeededMalformedConfigTests` (non-list custom, non-dict entries, non-dict overrides, malformed per-template override value). Full Python suite green (276), `ruff` clean.

File: app/renderer/src/routes/Processing.tsx [hunk 1]
```diff
@@ -109,6 +110,18 @@ export function Processing() {
           });
         }
       }),
+      ipc().on.processingProgress((e) => {
```
[the remark this answers is not in the document]
Comment (ruzin) [reply app/renderer/src/routes/Processing.tsx:113]: Addressed in 1e59188. The Processing screen shows the fresh-recording queue job, which emits bare `{line}` progress (no `summaryFile`), whereas a concurrent reprocess/report emits `summaryFile`-scoped progress (main.js 1929/2050). Added an `if (e.summaryFile) return;` guard so another meeting's scoped progress can't hijack this screen's stage — matching the `ProcessingProgressEvent` doc that scoped events exist so other views can ignore them.

File: src/report_store.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    return p.with_name(p.stem + "_reports.json")
+
+
+def load_sidecar(meeting_path) -> dict:
+    sp = sidecar_path(meeting_path)
+    if sp.exists():
+        try:
+            data = json.loads(sp.read_text(encoding="utf-8"))
+            if isinstance(data, dict):
+                data.setdefault("reports", [])
```
[the remark this answers is not in the document]
Comment (ruzin) [reply src/report_store.py]: Addressed in 1e59188. `load_sidecar` now normalizes key TYPES, not just presence: a non-list `reports` coerces to `[]` and a non-string `active_report` to `None`, so a malformed/hand-edited sidecar can't propagate and break downstream iteration/append. Added a regression test (`test_malformed_types_are_normalized`).
